spec/models/resource_spec.rb

```diff
@@ -38,6 +38,19 @@
     end
   end
 
+  context 'returns latest translation (published or not) for each language' do
```

Comment:
The best part about having these tests is that they should still pass after you refactor the `#latest` method.